plugins/modules/ec2_metadata_facts.py

```diff
@@ -576,6 +594,10 @@ def run(self):
         data.update(dyndata)
         data = self.fix_invalid_varnames(data)
 
+        instance_tags_keys = self._fetch(self.uri_instance_tags)
```

Comment:
My understanding is that tags are not enabled on instance metadata by default. Isn't this going to fail in that case?

Reply:
For the record, this is what I get with an instance and 2 tags:

```
        "ansible_ec2_tags_instance_Name": "test-goneri-tag",
        "ansible_ec2_tags_instance_foo": "var",
```

Reply:
@gravesm , it doesn't fail, it returns `str` "None" but I wasn't sure how to handle it

in case if the instance metadata tags are enabled, we are returning a `list` of tag keys like `"ansible_ec2_instance_tags_keys": [ "Name", "snake_case_key" ]` , so I decided to return `None` in a list like `"ansible_ec2_instance_tags_keys": [ "None" ]`, to maintain the output data type. 

should we want to return `ansible_ec2_instance_tags_keys: "None"` instead?

Reply:
> For the record, this is what I get with an instance and 2 tags:
>         "ansible_ec2_tags_instance_Name": "test-goneri-tag",
>         "ansible_ec2_tags_instance_foo": "var",

@goneri , these keys were being returned before the code change in this PR, but there was no single output param to access all the tag keys earlier
with the code change in this PR, we return new output param `ansible_ec2_instance_tags_keys` which is a list of str having tag keys

Reply:
> should we want to return ansible_ec2_instance_tags_keys: "None" instead?

I think we should return `ansible_ec2_instance_tags_keys: []`, which is what the documentation for the return value suggests it will do already.

Reply:
Fixed @gravesm

Reply:
Thank you @mandar242 for the clarification, I was far away.